dot/state/block.go

```diff
@@ -302,6 +307,15 @@ func (bs *BlockState) AddBlock(newBlock *types.Block) error {
 	return err
 }
 
+func (bs *BlockState) setLatestHeaderKey(hash common.Hash) error {
+	err := bs.db.db.Put(common.LatestHeaderHashKey, hash[:])
+	if err != nil {
```

Comment:
well, if err is nil, return err will also return nil. so the check is redundant;

Reply:
updated